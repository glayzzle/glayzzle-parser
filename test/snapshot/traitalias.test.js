const parser = require("../main");

describe("traitalias", function() {
  it("as public with method", function() {
    expect(
      parser.parseEval(`
class MyClass2 {
    use HelloWorld { sayHello as public myPrivateHello; }
}
`)
    ).toMatchSnapshot();
  });

  it("as method", function() {
    expect(
      parser.parseEval(`
class MyClass2 {
    use HelloWorld { sayHello as myPrivateHello; }
}
`)
    ).toMatchSnapshot();
  });

  it("as public", function() {
    expect(
      parser.parseEval(`
class MyClass1 {
    use HelloWorld { sayHello as public; }
}
`)
    ).toMatchSnapshot();
  });

  it("as protected", function() {
    expect(
      parser.parseEval(`
class MyClass1 {
    use HelloWorld { sayHello as protected; }
}
`)
    ).toMatchSnapshot();
  });

  it("as private", function() {
    expect(
      parser.parseEval(`
class MyClass1 {
    use HelloWorld { sayHello as private; }
}
`)
    ).toMatchSnapshot();
  });

  // PHP Fatal error:  Cannot use 'static' as method modifier
  // but should be parsable, because allowed by grammar
  it("as static", function() {
    const astErr = parser.parseEval(
      `
class MyClass1 {
    use HelloWorld { sayHello as static; }
}
`,
      {
        parser: {
          suppressErrors: true
        }
      }
    );
    expect(astErr).toMatchSnapshot();
  });

  // PHP Fatal error:  Cannot use 'abstract' as method modifier
  // but should be parsable, because allowed by grammar
  it("as abstract", function() {
    const astErr = parser.parseEval(
      `
class MyClass1 {
    use HelloWorld { sayHello as abstract; }
}
`,
      {
        parser: {
          suppressErrors: true
        }
      }
    );
    expect(astErr).toMatchSnapshot();
  });

  // PHP Fatal error:  Cannot use 'final' as method modifier
  // but should be parsable, because allowed by grammar
  it("as final", function() {
    const astErr = parser.parseEval(
      `
class MyClass1 {
    use HelloWorld { sayHello as final; }
}
`,
      {
        parser: {
          suppressErrors: true
        }
      }
    );
    expect(astErr).toMatchSnapshot();
  });

  it("as protected protected", function() {
    const astErr = parser.parseEval(
      `
class MyClass1 {
    use HelloWorld { sayHello as protected protected; }
}
`,
      {
        parser: {
          suppressErrors: true
        }
      }
    );
    expect(astErr).toMatchSnapshot();
  });
});
