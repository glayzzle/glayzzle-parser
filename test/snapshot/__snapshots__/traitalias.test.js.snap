// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`traitalias as abstract 1`] = `
Program {
  "children": Array [
    Class {
      "body": Array [
        TraitUse {
          "adaptations": Array [
            TraitAlias {
              "as": null,
              "kind": "traitalias",
              "method": "sayHello",
              "trait": null,
              "visibility": "",
            },
          ],
          "kind": "traituse",
          "traits": Array [
            ClassReference {
              "kind": "classreference",
              "name": "HelloWorld",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "extends": null,
      "implements": null,
      "isAbstract": false,
      "isAnonymous": false,
      "isFinal": false,
      "kind": "class",
      "name": Identifier {
        "kind": "identifier",
        "name": "MyClass1",
      },
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`traitalias as final 1`] = `
Program {
  "children": Array [
    Class {
      "body": Array [
        TraitUse {
          "adaptations": Array [
            TraitAlias {
              "as": null,
              "kind": "traitalias",
              "method": "sayHello",
              "trait": null,
              "visibility": "",
            },
          ],
          "kind": "traituse",
          "traits": Array [
            ClassReference {
              "kind": "classreference",
              "name": "HelloWorld",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "extends": null,
      "implements": null,
      "isAbstract": false,
      "isAnonymous": false,
      "isFinal": false,
      "kind": "class",
      "name": Identifier {
        "kind": "identifier",
        "name": "MyClass1",
      },
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`traitalias as method 1`] = `
Program {
  "children": Array [
    Class {
      "body": Array [
        TraitUse {
          "adaptations": Array [
            TraitAlias {
              "as": Identifier {
                "kind": "identifier",
                "name": "myPrivateHello",
              },
              "kind": "traitalias",
              "method": "sayHello",
              "trait": null,
              "visibility": "",
            },
          ],
          "kind": "traituse",
          "traits": Array [
            ClassReference {
              "kind": "classreference",
              "name": "HelloWorld",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "extends": null,
      "implements": null,
      "isAbstract": false,
      "isAnonymous": false,
      "isFinal": false,
      "kind": "class",
      "name": Identifier {
        "kind": "identifier",
        "name": "MyClass2",
      },
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`traitalias as private 1`] = `
Program {
  "children": Array [
    Class {
      "body": Array [
        TraitUse {
          "adaptations": Array [
            TraitAlias {
              "as": null,
              "kind": "traitalias",
              "method": "sayHello",
              "trait": null,
              "visibility": "private",
            },
          ],
          "kind": "traituse",
          "traits": Array [
            ClassReference {
              "kind": "classreference",
              "name": "HelloWorld",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "extends": null,
      "implements": null,
      "isAbstract": false,
      "isAnonymous": false,
      "isFinal": false,
      "kind": "class",
      "name": Identifier {
        "kind": "identifier",
        "name": "MyClass1",
      },
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`traitalias as protected 1`] = `
Program {
  "children": Array [
    Class {
      "body": Array [
        TraitUse {
          "adaptations": Array [
            TraitAlias {
              "as": null,
              "kind": "traitalias",
              "method": "sayHello",
              "trait": null,
              "visibility": "protected",
            },
          ],
          "kind": "traituse",
          "traits": Array [
            ClassReference {
              "kind": "classreference",
              "name": "HelloWorld",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "extends": null,
      "implements": null,
      "isAbstract": false,
      "isAnonymous": false,
      "isFinal": false,
      "kind": "class",
      "name": Identifier {
        "kind": "identifier",
        "name": "MyClass1",
      },
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`traitalias as protected protected 1`] = `
Program {
  "children": Array [
    Class {
      "body": Array [
        TraitUse {
          "adaptations": Array [
            TraitAlias {
              "as": Identifier {
                "kind": "identifier",
                "name": "protected",
              },
              "kind": "traitalias",
              "method": "sayHello",
              "trait": null,
              "visibility": "protected",
            },
          ],
          "kind": "traituse",
          "traits": Array [
            ClassReference {
              "kind": "classreference",
              "name": "HelloWorld",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "extends": null,
      "implements": null,
      "isAbstract": false,
      "isAnonymous": false,
      "isFinal": false,
      "kind": "class",
      "name": Identifier {
        "kind": "identifier",
        "name": "MyClass1",
      },
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`traitalias as public 1`] = `
Program {
  "children": Array [
    Class {
      "body": Array [
        TraitUse {
          "adaptations": Array [
            TraitAlias {
              "as": null,
              "kind": "traitalias",
              "method": "sayHello",
              "trait": null,
              "visibility": "public",
            },
          ],
          "kind": "traituse",
          "traits": Array [
            ClassReference {
              "kind": "classreference",
              "name": "HelloWorld",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "extends": null,
      "implements": null,
      "isAbstract": false,
      "isAnonymous": false,
      "isFinal": false,
      "kind": "class",
      "name": Identifier {
        "kind": "identifier",
        "name": "MyClass1",
      },
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`traitalias as public with method 1`] = `
Program {
  "children": Array [
    Class {
      "body": Array [
        TraitUse {
          "adaptations": Array [
            TraitAlias {
              "as": Identifier {
                "kind": "identifier",
                "name": "myPrivateHello",
              },
              "kind": "traitalias",
              "method": "sayHello",
              "trait": null,
              "visibility": "public",
            },
          ],
          "kind": "traituse",
          "traits": Array [
            ClassReference {
              "kind": "classreference",
              "name": "HelloWorld",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "extends": null,
      "implements": null,
      "isAbstract": false,
      "isAnonymous": false,
      "isFinal": false,
      "kind": "class",
      "name": Identifier {
        "kind": "identifier",
        "name": "MyClass2",
      },
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`traitalias as static 1`] = `
Program {
  "children": Array [
    Class {
      "body": Array [
        TraitUse {
          "adaptations": Array [
            TraitAlias {
              "as": null,
              "kind": "traitalias",
              "method": "sayHello",
              "trait": null,
              "visibility": "",
            },
          ],
          "kind": "traituse",
          "traits": Array [
            ClassReference {
              "kind": "classreference",
              "name": "HelloWorld",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "extends": null,
      "implements": null,
      "isAbstract": false,
      "isAnonymous": false,
      "isFinal": false,
      "kind": "class",
      "name": Identifier {
        "kind": "identifier",
        "name": "MyClass1",
      },
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
